lib/adapters/test/testplane.ts

```diff
@@ -0,0 +1,54 @@
+import {TestplaneTestResultAdapter} from '../test-result/testplane';
+import {TestStatus, UNKNOWN_ATTEMPT} from '../../constants';
+
+import type {TestAdapter} from './';
+import type {Test, Suite} from 'testplane';
+import type {ReporterTestResult} from '../test-result';
+
+export class TestplaneTestAdapter implements TestAdapter {
+    private _test: Test;
+
+    static create<T extends TestplaneTestAdapter>(this: new (test: Test) => T, test: Test): T {
+        return new this(test);
+    }
+
+    constructor(test: Test) {
+        this._test = test;
+    }
+
+    get originalTest(): Test {
+        return this._test;
+    }
+
+    get id(): string {
+        return this._test.id;
+    }
+
+    get pending(): boolean {
+        return this._test.pending;
+    }
+
+    get disabled(): boolean {
+        return this._test.disabled;
+    }
+
+    get browserId(): string {
+        return this._test.browserId;
+    }
+
+    fullTitle(): string {
+        return this._test.fullTitle();
+    }
+
+    clone(): TestplaneTestAdapter {
+        return TestplaneTestAdapter.create(this._test.clone());
+    }
+
+    isSilentlySkipped(runnable: Test | Suite = this._test): boolean {
```

Comment:
Is it possible to get rid of all methods and keep only `readonly` props? I believe it makes things much more simple and stable.

Not critical

Reply:
> Is it possible to get rid of all methods and keep only readonly props? I believe it makes things much more simple and stable.

I don't understand why methods in class should lead to some problems? in this case, I will have to write the same code inside helpers. And most of these helpers will have matching conditions for the name of the tool and it's much harder to support to my mind.